### PR TITLE
Added user input checks

### PR DESCRIPTION
**Pull request by aovick**
Added the getInput() method that asks for input until it's valid. Also, a while loop in main that ensures the number of reveals isn't greater than the number of coins.

### CODE REVIEW

**Comment by ajnugent** [comment]
Approved!

File: Coins.java [hunk 1]
```diff
@@ -23,4 +25,34 @@ public static void main(String args[]){
...
+                if(input <= 0)
+                    System.out.println("\nInvalid input: The number must be greater than zero.");
+                else
+                    return input;
+
+            } catch(Exception e) {
+                //Have Scanner consume the invalid input
+                in.next();
+                System.out.println("\nInvalid input: Not a number.");
+            }
```
Comment (zchampion) [inline Coins.java:55]: Clever way to handle that!

**Comment by zchampion** [comment]
LGTM :)

File: Coins.java (hunk 1 above)
Comment (aovick) [reply Coins.java:55]: Definitely the most important line. Would cause an infinite loop without it.

**Comment by dantanxiaotian** [comment]
Approved!

**Pull request merged**